packages/cozy-sharing/src/components/ShareByLink.jsx

```diff
@@ -66,47 +77,163 @@ class ShareByLink extends React.Component {
     }
   }
 
+  toggleMenu = () => {
+    this.setState(state => ({ ...state, menuIsOpen: !state.menuIsOpen }))
+  }
+
   render() {
     const t = this.context.t
-    const { loading } = this.state
-    const { checked, documentType } = this.props
+    const { loading, menuIsOpen } = this.state
+    const {
+      checked,
+      document,
+      documentType,
+      permissions,
+      onChangePermissions
+    } = this.props
+    const permissionCategories = get(
+      permissions,
+      '[0].attributes.permissions',
+      {}
+    )
+    const hasWritePermissions =
+      Object.values(permissionCategories).filter(permissionCategory =>
+        get(permissionCategory, 'verbs', []).includes('POST')
+      ).length > 0
```

Comment:
can we extract this logic from the render? And having a better place then the render to read it?

Reply:
I extracted the logic but kept the reading in the render, as I think it's the right place because It's an information derived from a prop. Do you think it would be better elsewhere?